File: src/main/python/com/hazza/LeetCodeByPython/SegmentOne/No26_Remove_Duplicates_from_Sorted_Array.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019/4/22 20:22
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums or len(nums) == 0:
            return 0

        cnt = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                nums[cnt] = nums[i]
                cnt += 1
            i += 1

        return cnt


if __name__ == '__main__':
    solution = Solution()
    assert solution.removeDuplicates([1, 1, 2]) == 2
    assert solution.removeDuplicates([0, 0, 1, 1, 1, 2, 2, 3, 3, 4]) == 5

File: src/main/java/com/hazza/leetcode/SegmentOne/No26RemoveDuplicatesFromSortedArray.java
package com.hazza.leetcode.SegmentOne;

/**
 * Created by hazza on 7/13/17.
 */
public class No26RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        int len = nums.length;
        int count  = 1;
        for (int i = 1; i < len; i++)
            if (nums[i] != nums[i - 1]) nums[count++] = nums[i];

        return count;
    }
}
